keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt

```diff
@@ -123,21 +127,46 @@ class ImageService(
    * Returns the latest image that is present in all regions.
    * Each ami must have tags.
    */
-  suspend fun getLatestNamedImageWithAllRegions(packageName: String, account: String, regions: List<String>): NamedImage? =
-    cloudDriverService.namedImages(
+  suspend fun getLatestNamedImageWithAllRegions(resourceId: ResourceId, packageName: String, account: String, regions: List<String>): NamedImage? {
+    val images = cloudDriverService.namedImages(
       serviceAccount = DEFAULT_SERVICE_ACCOUNT,
       imageName = packageName,
       account = account
     )
+
+    val filteredImages = images
       .filter { it.hasAppVersion }
       .sortedWith(NamedImageComparator)
+
+    val eliminatedImages = mutableMapOf<String, String>()
+    val image = filteredImages
       .find {
+        var errors = ""
         val curAppVersion = AppVersion.parseName(it.appVersion)
-        curAppVersion.packageName == packageName &&
-          it.accounts.contains(account) &&
-          it.amis.keys.containsAll(regions) &&
-          tagsExistForAllAmis(it.tagsByImageId)
+        if (curAppVersion.packageName != packageName) {
+          errors += "[package name does not match, package is ${curAppVersion.packageName}]"
+        }
+        if (!it.accounts.contains(account)) {
+          errors += "[image is not in account $account - it's in accounts ${it.accounts}]"
+        }
+        if (!it.amis.keys.containsAll(regions)) {
+          errors += "[image is only in regions ${it.amis.keys}, not all required regions ($regions)]"
+        }
+        if (!tagsExistForAllAmis(it.tagsByImageId)) {
+          errors += "[image does not have tags for all regions ($regions): existing tags ${it.tagsByImageId}]"
+        }
+        if (errors == "") {
+          true
+        } else {
+          eliminatedImages[it.imageName] = errors
+          false
+        }
       }
+    publisher.publishEvent(DebugEvent(resourceId, "Finding latest qualifying named image for $packageName", "rejected images=$eliminatedImages"))
```

Comment:
Why are there 2 events published here?

Reply:
I want to log all the images that were rejected and why, as well as the image that was chosen